cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -34,6 +34,16 @@ private class IteratorByTraits extends Iterator {
   IteratorByTraits() { exists(IteratorTraits it | it.getIteratorType() = this) }
 }
 
+/**
+ * The C++ standard includes an `std::iterator_traits` specialization for pointer types. When
+ * this specialization is included in the database, a pointer type `T*` will be an instance
+ * of the `IteratorByTraits` class. However, if the `T*` specialization is not in the database,
+ * we need to explicitly include them with this class.
+ */
+private class IteratorByPointer extends Iterator instanceof PointerType {
+  IteratorByPointer() { not this instanceof IteratorByTraits }
+}
```

Comment:
This LGTM.

Do you have any idea what the downside would be if we made something an `Iterator` when it shouldn't be?  e.g. if we made every `IntegralType` an `Iterator`, would we get some kind of incorrect flow through integer parameters?

Reply:
Yeah, I think such a change would mean that we get flow in cases like:
```
void f(int n) {
  n = source();
}

void test() {
  int n = 0;
  f(n);
  sink(n);
}
```
I'm not sure if we've set up everything such that this was the only thing necessary to get flow in such cases, but it would certainly be a likely outcome of such a change.